src/shared/conf-parser.c

```diff
@@ -1398,21 +1398,48 @@ int config_parse_compression(
                 void *data,
                 void *userdata) {
 
-        Compression *compression = ASSERT_PTR(data);
+        CompressionOpts ***opts = ASSERT_PTR(data);
+        int level = -1;
+        char *l = NULL;
+        _cleanup_free_ char *t = NULL;
 
         if (isempty(rvalue)) {
-                *compression = COMPRESSION_NONE;
                 return 1;
         }
 
-        Compression c = compression_lowercase_from_string(rvalue);
+        l = strchr(rvalue, ':');
+        if (l) {
+                double lVal = strtol(l, NULL, 10);
+                if (lVal < 0)
+                        return log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                          "Compression level %s should be positive", l);
```

Comment:
if you ignore invalid settings (which is often ok), then please say so by suffixing the log message with ", ignoring."